matron/src/device/device_list.c

```diff
@@ -50,6 +51,7 @@ void dev_list_init(void) {
     dq.size = 0;
     dq.head = NULL;
     dq.tail = NULL;
+    pthread_mutex_init(&dq.lock, NULL);
```

Comment:
Might consider using a static initializer for the mutex, or check the result of `pthread_mutex_init()` for success.